docs/excel/custom-functions-best-practices.md

```diff
@@ -0,0 +1,184 @@
+        }
+
+        // In this simple example we always resolve, but you should also add error handling for reject
+        _batch[i].resolve(sum);
+      }
+
+      // The current batch has been processed
+      _batch = [];
+      _isBatchScheduled = false;
+    }
+  }
+  
+  
+  // Sample usage
+  console.clear();
+  sumAsync(1, 2).then((value) => { console.log("1 + 2 = " + value); });
+  sumAsync(2, 4, 6).then((value) => { console.log("2 + 4 + 6 = " + value); });
+  sumAsync(3, 5, 7, 9).then((value) => { console.log("3 + 5 + 7 + 9 = " + value); });
+```
+
+## Error handling
+
+Error handling for custom functions is the same as [error handling for the Excel JavaScript API at large](./excel-add-ins-error-handling.md). Generally, you will use `.catch` to handle errors. The code below gives an example of `.catch`.
+
+```js
+function getComment(x) {
+    var url = "https://jsonplaceholder.typicode.com/comments/" + x; //this delivers a section of lorem ipsum from the jsonplaceholder API
+    return fetch(url)
+        .then(function (data) {
+            return data.json();
+        })
+        .then((json) => {
+            return json.body;
+        })
+        .catch(function (error) {
+            throw error;
+        })
+}
+```
+
+## Error logging
+
+You can use runtime logging to debug your custom function's XML manifest file or to look for errors in your custom functions in real time via console.log statements. Runtime logging is only available for Office 2016 desktop currently.
+
+For full instructions on how to use runtime logging, [read this article](../testing/troubleshoot-manifest.md).
+
+## Debugging
+
+Development on debugging tools is still proceeding for custom functions, which are still in preview.  
+
+At present, the best method for debugging is to use [Excel through Office Online](https://www.office.com/launch/excel) and use the F12 debugging tool native to your browser.
+
+## Mapping names
+
+Custom functions are typically declared entirely in uppercase letters, although you can change this using the `CustomFunctionsMappings` object. Mapping functions in this object allows you to create any name for the function, which you can then use when you call it in Excel.  
+  
+You can declare individual functions, as shown below:  
+
+```js
+function ADD42(num) {
+    return num + 42;  
+}  
+  
+CustomFunctionsMappings = {
+    "plusFortyTwo" : ADD42 //effectively renames the add-in when invoked in Excel, so you will now call =plusFortyTwo()
+}
+```
+
+However, you can declare multiple mappings at the same time, as shown in the example below.  
+
+```js
+//assume that COUNTDOGS and COUNTCATS exist
+  
+CustomFunctionsMappings = {
+    "countdogs" : COUNTDOGS,  
+    "meow" : COUNTCATS //the mapping is not required to be the same
+}
+ ```
+
+The benefits of using `CustomFunctionsMappings` include:  
+
+- You can use web tools like webpack, as well as import syntax (commonly used in React, for example).
```

Comment:
This bullet needs a lot of explanation. What is the relation between webpack and custom function mappings? What would go wrong with webpack if we didn't have customfunctionmappings?